frontend/src/shared/util/rich-text-utils.tsx

```diff
@@ -1,177 +1,220 @@
+    }, [editor]);
 
-                    if (buttonLabel) {
-                        button.setAttribute('aria-label', buttonLabel);
-                    }
-                    if (imageLabel) {
-                        addTitleNodeIfNeeded(button, imageLabel);
-                    }
-                }
-            }
+    const setLink = useCallback(() => {
+        const previousUrl = editor?.getAttributes('link').href;
+        const url = window.prompt('URL', previousUrl);
 
-            const colorPicker = toolbar.querySelector('.ql-color-picker');
-            if (colorPicker) {
-                const label = colorPicker.getElementsByClassName('ql-picker-label')[0];
-                const optionsContainer = colorPicker.getElementsByClassName('ql-picker-options')[0];
-                const options = optionsContainer.getElementsByClassName('ql-picker-item');
+        // canceled
+        if (url === null) {
+            return;
+        }
 
-                label.setAttribute('role', 'button');
-                label.setAttribute('aria-haspopup', 'true');
-                label.setAttribute('tabIndex', '0');
-                label.setAttribute('aria-label', 'Set text color');
+        // empty
+        if (url === '') {
+            editor?.chain().focus().extendMarkRange('link').unsetLink().run();
 
-                optionsContainer.setAttribute('aria-hidden', 'true');
-                optionsContainer.setAttribute('aria-label', 'submenu');
+            return;
+        }
 
-                for (let x = 0; x < options.length; x++) {
-                    const item = options[x];
+        // update link
+        editor
+            ?.chain()
+            .focus()
+            .extendMarkRange('link')
+            .setLink({ href: url })
+            .run();
+    }, [editor]);
 
-                    // Read the css 'content' values and generate aria labels
-                    item.setAttribute('aria-label', item['dataset'].value);
-                    item.addEventListener('keyup', (e) => {
-                        if (e.keyCode === 13) {
-                            item.click();
-                            optionsContainer.setAttribute('aria-hidden', 'true');
-                        }
-                    });
-                }
+    if (!editor) {
+        return null;
+    }
 
-                label.addEventListener('keyup', (e) => {
-                    if (e.keyCode === 13) {
-                        label.click();
-                        optionsContainer.setAttribute('aria-hidden', 'false');
-                    }
-                });
-            }
-        });
-    }, []);
+    editor.on('update', ({ editor }) => {
+        props.onChange(editor.getHTML());
+    });
 
     return (
-        <ReactQuill
-            modules={modules}
-            formats={formats}
-            value={props.value}
-            onChange={props.onChange}
-        />
+        <Container>
+            <ColumnLayout columns={1} borders='horizontal'>
```

Comment:
Why are we using columns=1? What is this ColumnLayout doing that wouldn't normally occur without it?

Reply:
Columns=1 is forcing all the buttons to stay in a single horizontal line. otherwise, the buttons would be separated in certain screens